### ISSUE TITLE
Advanced Scheduler-Options

### ISSUE THREAD

**Issue by warmup72** [borgbase/vorta#1403]
It would be nice to add the usual options to the scheduler in Vorta:
1. Selection of the day(s) of the week via checkboxes
2. Selection of the start time for a periodic run
3. Option that only [n] backup plans are allowed to run in parallel
4. Exclude a certain period of time within a day (e.g. no backup between 10:00 p.m. - 6:00 a.m.)

**Comment by real-yfprojects** [comment]
>     * [ ]  Option that only [n] backup plans are allowed to run in parallel

In case of multiple profiles scheduled for the same time?

**Comment by warmup72** [comment]
I think it's important to specify the maximum number of jobs that can run in parallel at the same time. also across profiles

**Comment by real-yfprojects** [comment]
Currently backups of the same profile cannot run in parallel. But your point applies for backups from different profiles.

**Comment by Znerox** [comment]
Having the option to do backup every X weeks, on a specific weekday and time would be very nice.

**Comment by real-yfprojects** [comment]
> Having the option to do backup every X weeks, on a specific weekday and time would be very nice.

What do you need that for?

**Comment by thatso** [comment]
More than one year later and there is still no possibility to select specific weekdays for schedules. :-1: 
I was surprised to realize that Vorta does not offer such basic scheduling options despite the fact that corresponding pull requests like [this](https://github.com/borgbase/vorta/issues/551#issuecomment-734535053) exist since 2020.

**Comment by real-yfprojects** [comment]
@thatso As this is a community maintained project and not a commercial product, it might need a motivated individual like you to pick up the issue. 

Also, my question still remains unanswered: What is a use case for selecting specific weekdays for schedules?

**Comment by warmup72** [comment]
Hello,
In my opinion, this is basically an easy way to cover all requirements.
However, the problem with Vorta in particular is that you cannot select a time if you select a backup interval greater than one day. In this case Vorta takes the time of the last backup. Quite annoying for me.

**Comment by real-yfprojects** [comment]
> despite the fact that corresponding pull requests like [this](https://github.com/borgbase/vorta/issues/551#issuecomment-734535053) exist since 2020.

The PR's contents no longer exists, so this isn't really an argument.

Nevertheless I feel like a GUI like following would actually be realively easy to implement and provide sufficient UX. However instead of changing the visibility of the UI elements they should be disabled/enabled as needed. (They are still displayed but greyed out)

![100402946-22c11580-3023-11eb-9443-b209e1c9133e](https://github.com/borgbase/vorta/assets/62463991/e58efe51-2fba-4934-9885-8c10e05dd1d8)

**Comment by thatso** [comment]
> Nevertheless I feel like a GUI similar to the following would actually be realively easy to implement and provide sufficient UX.

That UI is exactly the one from the pull request I referenced.
The specific way the selection will be implemented is not really relevant. However, not being able to choose weekdays and times at all is just unexpected as this is a well-known and acknowledged standard for any backup software in general.

**Comment by Znerox** [comment]
The reason I would want something like this: I only want backups to run at night, and for many of my backup jobs, the source folder changes so rarely that having it run once a week would be sufficient.
Unfortunately I do not know enough to implement this myself.

**Comment by b3rrytech** [comment]
> Also, my question still remains unanswered: What is a use case for selecting specific weekdays for schedules?

I'll add my reasoning as well. I have multiple physical remote backup repositories that only run at certain hours/days due to both location, hardware sound and bandwidth issues. Since Borg doesn't really play well with pull backups I prefer to schedule the backups to the times I know my remote repositories are reachable. Yes, it could probably be done by adding a pre-backup script that pings the remote IP so that the backup never runs if the remote IP isn't up. But a more granular control would be a much cleaner and simple solution.

I'm no programmer so I might be off here, but what about a custom field for a cron expression? Or doesn't that work with Qtimer? This would keep the UI more or less "as is" and would add the option for very granular control if needed.

**Comment by thatso** [comment]
Unbelievable. Another year has passed and this issue is still open. 😠 
@real-yfprojects: the use case is called weekly backups. Simple as that. But you already knew that and just enjoy trolling.

**Comment by Znerox** [comment]
> Unbelievable. Another year has passed and this issue is still open. 😠 [@real-yfprojects](https://github.com/real-yfprojects): the use case is called weekly backups. Simple as that. But you already knew that and just enjoy trolling.

No matter how useful this feature would be, it's a community maintained project. Demands are not productive. 🙂

**Comment by thatso** [comment]
Neither are programmers in their high tower refusing to implement standard functionality.

**Comment by m3nu** [comment]
Which of those are general enough to add and don't need too much new code, @VandalByte ?

We also don't want to add very exotic options that  aren't commonly found in other tools.

**Comment by VandalByte** [comment]
Most requested is the time to start a backup and the possibility of running backup on weekends and on specific days.
So considering minimal code changes, an addition of something like this should be enough:

![Image](https://github.com/user-attachments/assets/d06d75c6-8035-4caf-bc86-ce1d1fdec963)

It solves the need for backup to start at a specific time. Then the weekend problem also can be solved with this if user sets the interval as `Interval: 1 Week 03:32` on a Saturday, next backup will be on next Saturday, it's not perfect but this will get the job done.

**Comment by VandalByte** [comment]
Another idea I had was adding check boxed like `[ ] Mon  [ ] Tue etc..` with a new radio `Backup on` and giving time there, so basically you can select which day and which time you want the backup to run, but will require quite a bit of new addition.

So something like this:

![Image](https://github.com/user-attachments/assets/2b32cbf0-160f-40ee-b24b-8c12cfeefb9a)

**Comment by warmup72** [comment]
Looky nice

**Comment by m3nu** [comment]
Yeah, good UI. Reminds me a bit of Android alarm clock app. They also have days of the week to check.

**Comment by VandalByte** [comment]
Hey guys, so for everyone who's been following this issue, I would like your opinion on this design concept:

![Image](https://github.com/user-attachments/assets/42e4aafe-6410-4fb7-a36d-0dcfa1996fb4)

So this would include the addition of a new option as shown, which will allow you to select which day of the week to backup on and on what time. This will hopefully solve most of the issues you guys mentioned here. I haven't started the development yet, I thought I should discuss it with everyone first before starting. 

One other thing to consider is that this addition will make the `Backup daily` option redundant, as we can just select all the checkboxes to do the same.

Otherwise the simplest option is the one mentioned [here](https://github.com/borgbase/vorta/issues/1403#issuecomment-2882038350), but this is more like a patch work rather than a solution.

**Comment by m3nu** [comment]
Would the day of week and and allowed time be a limitation on the periodic option mainly? Together with limiting it to some hours within a day.

Also, just to remind that deciding to keep the current setup is an option too. There is no pressure to pick some other option, if there is no clear benefit and it's easy to understand. This decision and the way to it is as valuable as adding new features. Since we already support starting backups from the CLI, it's easy enough to use any Cron software to start a backup.

**Comment by VandalByte** [comment]
> Would the day of week and and allowed time be a limitation on the periodic option mainly? Together with limiting it to some hours within a day.

The [last PR](https://github.com/borgbase/vorta/pull/551#issuecomment-734535053) we had was something similar. Few problems I can think of with that approach are:

- When that PR was made we didn't had any options on the side, but with the current side by side layout it will just add too much options and UI might look a bit off.
- Also if we add it that way then it means adding a choice box which limits the backup to a single specific day of the week. It's not much useful because, For example if I wanted to do the backup on just a specific day of the week, I would just wait for that day and turn on periodic backup every week.

So as far as I think the periodic option is perfect as of right now.

> Since we already support starting backups from the CLI, it's easy enough to use any Cron software to start a backup.

Yeah or we can do this.

**Comment by warmup72** [comment]
In my option it is normal for a Backuptool to offer this Choice-Option.
Sure you can wait for this day or use a cron-job. But is this a handy, usefull app-style?

Another idea - in my case i want to run borg not daily, not weekly, but only some times a week.

**Comment by m3nu** [comment]
> In my option it is normal for a Backuptool to offer this Choice-Option.

Could you share some screenshots of how other tools solve this? So we can take some inspiration from there?

Here how Arq does it:

![Image](https://github.com/user-attachments/assets/38ec86c6-a397-4e9a-8098-abaca448a417)
